jaeger-core/src/main/java/io/jaegertracing/internal/propagation/TextMapCodec.java

```diff
@@ -125,24 +125,34 @@ public JaegerSpanContext extract(TextMap carrier) {
           baggage = new HashMap<String, String>();
         }
         baggage.put(keys.unprefixedKey(key, baggagePrefix), decodedValue(entry.getValue()));
+      } else if (key.equals(Constants.BAGGAGE_HEADER_KEY)) {
+        baggage = parseBaggageHeader(decodedValue(entry.getValue()), baggage);
       }
     }
-    if (context == null) {
-      if (debugId != null) {
-        return objectFactory.createSpanContext(
-            0,
-            0,
-            0,
-            (byte) 0,
-            Collections.<String, String>emptyMap(),
-            debugId);
-      }
-      return null;
-    }
-    if (baggage == null) {
+    if (debugId == null && baggage == null) {
       return context;
     }
-    return context.withBaggage(baggage);
+    return objectFactory.createSpanContext(
+      context == null ? 0 : context.getTraceId(),
+      context == null ? 0 : context.getSpanId(),
+      context == null ? 0 : context.getParentId(),
+      context == null ? (byte)0 : context.getFlags(),
+      baggage,
+      debugId);
+  }
+
+  private Map<String, String> parseBaggageHeader(String header, Map<String, String> baggage) {
+    String[] parts = header.split("\\ *,\\ *");
```

Comment:
Programmatically set baggage can have `,` or `=` in the baggage value while it seems adhoc baggage cannot.  While it is obvious, we should call out this limitation in documentation.

Reply:
That seems kind of obvious if you're asked to provide baggage in the `k1=v1, k2=v2` format.